### ISSUE TITLE
Ensure no warnings in documentation builds

### ISSUE THREAD

**Issue by AyodeAwe** [rapidsai/cudf#12711]
**Is your feature request related to a problem? Please describe.**
Currently, the `cudf` documentation does not build without some issues. These issues show up as warnings in the documentation build logs. An example is here:

```
Warning, treated as error:
[autosummary] failed to import cudf.Index.argsort.
Possible hints:
* AttributeError: type object 'Index' has no attribute 'argsort'
* ModuleNotFoundError: No module named 'cudf.Index'
```

This was discovered when attempting to run the sphinx-build command with the `-W` flag. The `-W` flag causes the build to fail in the event of warnings by turning warnings to errors. (See [here](https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-W))

**Describe the solution you'd like**
The ideal scenario is to have the `cudf` documentation build successfully without warnings. The addition of the `-W` flag to the sphinx build command would ensure this.

**Describe alternatives you've considered**
We currently run the build command without the `-W` flag, ignoring any warnings that get outputted.

**Additional context**
[This PR](https://github.com/rapidsai/cudf/pull/12592) contains commits with the `-W` flag added.

**Comment by AyodeAwe** [comment]
cc: @ajschmidt8 @bdice for viz

**Comment by bdice** [comment]
Fixing this depends on some of the changes discussed here: https://github.com/rapidsai/cudf/pull/12693#issuecomment-1416509632

**Comment by vyasr** [comment]
Fixed in #13789
